### DISCUSSION TITLE
liveAnnouncer direct DOM manipulation and issues with react-testing-library

### DISCUSSION THREAD

**Discussion by alirezamirian** [adobe/react-spectrum#4915] [General]
We faced an issue in some test suites written using react-testing-library, and it seems not like an issue specific to our setup, so sharing it here. 
The issue was test cases affecting each other, and after lots of tracing, and debugging, it turned out it was due to liveAnnouncer leaving behind some role="log" elements, with a content that was matching some of our test selectors in the subsequent test cases. 
After looking at LiveAnnouncer code, we realized it directly manipulates DOM, for some technical reason:
https://github.com/adobe/react-spectrum/blob/73d8be4dd7ff1713084cfe3f602f11aef1cecb07/packages/%40react-aria/live-announcer/src/LiveAnnouncer.tsx#L54-L59

The issue can be worked around, by calling `destroyAnnouncer` in a beforeEach block, (or just finding all role="logs" and removing them), but it's a very obscure issue, and maybe you want to [document it](https://react-spectrum.adobe.com/react-spectrum/testing.html) or consider changing the live announcer implementation to be react-aware, despite the explained challenges regarding react-dom 18 API differences.

**Comment by reidbarber** [comment]
@LFDanLu has an [RFC](https://github.com/adobe/react-spectrum/pull/4835#top) for some testing utilities, so this seems like a great fit for there (or in our docs until then.)
